### ISSUE TITLE
Facebook ANE 4.30.0 does not login on iOS (works on Android) - URGENT

### ISSUE THREAD

**Issue by PippoApps** [myflashlab/facebook-ANE#126]
## Environments ##
* iOS Version: 9.3.5 and 10.3.3
* AIR SDK Version: 28 and 29

## Problem Description ##

- Same exact configuration and code works perfectly on Android.
- Used ANELab upgrading from previous project using last available Facebook ANE
- Same manifest file for iOS and Android
- No Facebook APP installed
- User already logged in in Safari

On iOS, calling 
```
Facebook.init(_appId);
Facebook.listener.addEventListener(FacebookEvents.INIT, onAneInit);
Facebook.listener.addEventListener(FacebookEvents.INVOKE, onAneInvoke);
```
Does not call any callback. 
Only output in console is this (through Overrideair):
```
FB_ANE> checkOverride_openURLOptions
FB_ANE> com.yappingmoose.buncobonko://google/link/?dismiss=1&is_weak_match=1
```

## Steps to Reproduce ##
Just call Facebook.init(appId);

If user is not logged in in Safari, callbacks are not triggered but there is no console output.

**Comment by PippoApps** [comment]
It's not really consistent. On Apple console reports this:
Apr 18 22:51:04 Filippos-iPad Bunco Bonko[958] <Warning>: FB_ANE|||BaseFB|||checkOverride_openURLOptions
Apr 18 22:51:04 Filippos-iPad Bunco Bonko[958] <Warning>: FB_ANE|||BaseFB|||com.yappingmoose.buncobonko://google/link/?dismiss=1&is_weak_match=1
Apr 18 22:51:04 Filippos-iPad Bunco Bonko[958] <Warning>: FB_ANE> checkOverride_openURLOptions

**Comment by PippoApps** [comment]
Consistently, after a few minutes a crash is reported, even though AIR app interface continues to be responsive:
```
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Notice>: platform_thread_get_unique_id matched 1261485
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Notice>: Formulating report for process[963] Bunco Bonko
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Warning>: Report of type '185(185_Bunco Bonko)' not saved because the limit of 25 logs has been reached
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Notice: This report is abbreviated for syslog inclusion because it could not be saved to disk.
	Symbolication may be possible by manually cleaning up and including the Binary Image section of a full report from this same device -- good luck!
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Process:             Bunco Bonko [963]
	Path:                /var/containers/Bundle/Application/B41007ED-2918-4556-ABC9-533D0ECF27C3/Bunco Bonko.app/Bunco Bonko
	OS Version:          iOS 9.3.5 (13G36)
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Exception Type:  EXC_RESOURCE
	Exception Subtype: WAKEUPS
	Exception Message: (Limit 150/sec) Observed 0/sec over 300 secs
	Exception Note:  NON-FATAL CONDITION (this is NOT a crash)
	Triggered by Thread:  0
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 0 name:  Dispatch queue: com.apple.main-thread
	Thread 0 Attributed:
	0   Bunco Bonko                   	0x0057d1e8 0xad000 + 5046760
	1   Bunco Bonko                   	0x0020705c 0xad000 + 1417308
	2   Bunco Bonko                   	0x001e4244 0xad000 + 1274436
	3   Bunco Bonko                   	0x00206644 0xad000 + 1414724
	4   Bunco Bonko                   	0x001ca530 0xad000 + 1168688
	5   Bunco Bonko                   	0x001dcd08 0xad000 + 1244424
	6   Bunco Bonko                   	0x00160aa0 0xad000 + 735904
	7   Bunco Bonko                   	0x0019d390 0xad000 + 983952
	8   Bunco Bonko                   	0x001a65e8 0xad000 + 1021416
	9   Bunco Bonko                   	0x001a6844 0xad000 + 1022020
	10  Bunco Bonko                   	0x001a6cb4 0xad000 + 1023156
	11  Bunco Bonko                   	0x00161a20 0xad000 + 739872
	12  Bunco Bonko                   	0x0019d390 0xad000 + 983952
	13  Bunco Bonko                   	0x001a65e8 0xad000 + 1021416
	14  Bunco Bonko                   	0x001a6844 0xad000 + 1022020
	15  Bunco Bonko                   	0x001a6d68 0xad000 + 1023336
	16  Bunco Bonko                   	0x001ee644 0xad000 + 1316420
	17  Bunco Bonko                   	0x001efc2c 0xad000 + 1322028
	18  Bunco Bonko                   	0x000c77c0 0xad000 + 108480
	19  QuartzCore                    	0x25357a0a 0x252fd000 + 371210
	20  QuartzCore                    	0x25357856 0x252fd000 + 370774
	21  IOMobileFramebuffer           	0x24a3e50a 0x24a39000 + 21770
	22  IOKit                         	0x22f4c758 0x22f48000 + 18264
	23  CoreFoundation                	0x22caa37c 0x22c02000 + 689020
	24  CoreFoundation                	0x22cbe5b2 0x22c02000 + 771506
	25  CoreFoundation                	0x22cbdcc6 0x22c02000 + 769222
	26  CoreFoundation                	0x22cbc0d8 0x22c02000 + 762072
	27  CoreFoundation                	0x22c0b228 0x22c02000 + 37416
	28  CoreFoundation                	0x22c0b014 0x22c02000 + 36884
	29  GraphicsServices              	0x241fbac8 0x241f2000 + 39624
	30  UIKit                         	0x272df188 0x27263000 + 508296
	31  Bunco Bonko                   	0x0022f998 0xad000 + 1583512
	32  libdyld.dylib                 	0x228b3872 0x228b1000 + 10354
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Activity ID:             0x00000000001ab996
	Activity Name:           starting resolver activity
	Activity Image Path:     /usr/lib/system/libsystem_network.dylib
	Activity Image UUID:     E71DD2FC-95A3-314B-A344-9784128E0EFB
	Activity Offset:         0x00028c33
	Activity Running Time:   110.863823 sec
	
	Trace Messages (reverse chronological seconds):
	21.205668    Bunco Bonko               	0x0000000022ae5db1 
	21.207132    Bunco Bonko               	0x0000000022ae5db1 
	21.281621    Bunco Bonko               	0x0000000022ae5db1 
	21.281911    Bunco Bonko               	0x0000000022ae5db1 
	104.877482   Bunco Bonko               	0x0000000022ae5db1 
	104.880161   Bunco Bonko               	0x0000000022ae5db1 
	104.880260   CFNetwork                 	0x000000002327c9f5 TCP Conn 0x15d1cd50 started
	108.355745   Bunco Bonko               	0x0000000022ae5db1 
	108.368332   Bunco Bonko               	0x0000000022ae5db1 
	108.368483   CFNetwork                 	0x000000002327c9f5 TCP Conn 0x15e583b0 started
	110.688775   Bunco Bonko               	0x0000000022ae5db1 
	110.688898   Bunco Bonko               	0x0000000022ae5db1 
	110.706250   Bunco Bonko               	0x0000000022ae5db1 
	110.711237   Bunco Bonko               	0x0000000022ae5db1 
	110.720063   Bunco Bonko               	0x0000000022ae5db1 
	110.725120   Bunco Bonko               	0x0000000022ae5db1 
	110.731303   Bunco Bonko               	0x0000000022ae5db1 
	110.732291   Bunco Bonko               	0x0000000022ae5db1 
	110.734734   Bunco Bonko               	0x0000000022ae5db1 
	110.735305   Bunco Bonko               	0x0000000022ae5db1 
	110.737771   Bunco Bonko               	0x0000000022ae5db1 
	110.740478   Bunco Bonko               	0x0000000022ae5db1 
	110.740771   Bunco Bonko               	0x0000000022ae5db1 
	110.741342   Bunco Bonko               	0x0000000022ae5db1 
	110.741401   CFNetwork                 	0x000000002327c9f5 TCP Conn 0x15e48be0 started
	110.741458   CFNetwork                 	0x000000002327c9f5 TCP Conn 0x15e49120 started
	110.741458   CFNetwork                 	0x000000002327c9f5 TCP Conn 0x15e49430 started
	110.768470   Bunco Bonko               	0x0000000022ae5db1 
	110.790202   Bunco Bonko               	0x0000000022ae5db1 
	110.792673   Bunco Bonko               	0x0000000022ae5db1 
	110.793250   CFNetwork                 	0x000000002327c9f5 TCP Conn 0x15d47dd0 started
	110.803874   CFNetwork                 	0x00000000232c215f Creating default cookie storage with default identifier
	110.803874   CFNetwork                 	0x00000000232c213b Faulting in CFHTTPCookieStorage singleton
	110.803874   CFNetwork                 	0x00000000233047bf Faulting in NSHTTPCookieStorage singleton
	110.806216   Bunco Bonko               	0x0000000022ae5db1 
	110.827807   Bunco Bonko               	0x0000000022ae5db1
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 1 name:  Dispatch queue: com.apple.libdispatch-manager
	Thread 1:
	0   libsystem_kernel.dylib        	0x229882f8 0x22972000 + 90872
	1   libdispatch.dylib             	0x2287dd60 0x22867000 + 93536
	2   libdispatch.dylib             	0x2287dabe 0x22867000 + 92862
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 2 name:  BackgroundThread
	Thread 2:
	0   libsystem_kernel.dylib        	0x22986998 0x22972000 + 84376
	1   libsystem_pthread.dylib       	0x22a2edec 0x22a2c000 + 11756
	2   libsystem_pthread.dylib       	0x22a2fc98 0x22a2c000 + 15512
	3   Bunco Bonko                   	0x0055d350 0xad000 + 4916048
	4   Bunco Bonko                   	0x0031f84c 0xad000 + 2566220
	5   Bunco Bonko                   	0x0055d028 0xad000 + 4915240
	6   Bunco Bonko                   	0x0055d16c 0xad000 + 4915564
	7   Bunco Bonko                   	0x0055cd08 0xad000 + 4914440
	8   libsystem_pthread.dylib       	0x22a2f85a 0x22a2c000 + 14426
	9   libsystem_pthread.dylib       	0x22a2f7ce 0x22a2c000 + 14286
	10  libsystem_pthread.dylib       	0x22a2d724 0x22a2c000 + 5924
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 3 name:  BackgroundThread
	Thread 3:
	0   libsystem_kernel.dylib        	0x22986998 0x22972000 + 84376
	1   libsystem_pthread.dylib       	0x22a2edec 0x22a2c000 + 11756
	2   libsystem_pthread.dylib       	0x22a2fc98 0x22a2c000 + 15512
	3   Bunco Bonko                   	0x0055d350 0xad000 + 4916048
	4   Bunco Bonko                   	0x0031f84c 0xad000 + 2566220
	5   Bunco Bonko                   	0x0055d028 0xad000 + 4915240
	6   Bunco Bonko                   	0x0055d16c 0xad000 + 4915564
	7   Bunco Bonko                   	0x0055cd08 0xad000 + 4914440
	8   libsystem_pthread.dylib       	0x22a2f85a 0x22a2c000 + 14426
	9   libsystem_pthread.dylib       	0x22a2f7ce 0x22a2c000 + 14286
	10  libsystem_pthread.dylib       	0x22a2d724 0x22a2c000 + 5924
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 4:
	0   libsystem_kernel.dylib        	0x22986ffc 0x22972000 + 86012
	1   libsystem_c.dylib             	0x228daedc 0x228d0000 + 44764
	2   Bunco Bonko                   	0x0055cd70 0xad000 + 4914544
	3   Bunco Bonko                   	0x002616c4 0xad000 + 1787588
	4   Bunco Bonko                   	0x00261680 0xad000 + 1787520
	5   Bunco Bonko                   	0x0055d028 0xad000 + 4915240
	6   Bunco Bonko                   	0x0055d16c 0xad000 + 4915564
	7   Bunco Bonko                   	0x0055cd08 0xad000 + 4914440
	8   libsystem_pthread.dylib       	0x22a2f85a 0x22a2c000 + 14426
	9   libsystem_pthread.dylib       	0x22a2f7ce 0x22a2c000 + 14286
	10  libsystem_pthread.dylib       	0x22a2d724 0x22a2c000 + 5924
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 5 name:  ScriptTimeout
	Thread 5:
	0   libsystem_kernel.dylib        	0x22986998 0x22972000 + 84376
	1   libsystem_pthread.dylib       	0x22a2edec 0x22a2c000 + 11756
	2   libsystem_pthread.dylib       	0x22a2fcd8 0x22a2c000 + 15576
	3   Bunco Bonko                   	0x0055d308 0xad000 + 4915976
	4   Bunco Bonko                   	0x0042e9a4 0xad000 + 3676580
	5   Bunco Bonko                   	0x0055d028 0xad000 + 4915240
	6   Bunco Bonko                   	0x0055d16c 0xad000 + 4915564
	7   Bunco Bonko                   	0x0055cd08 0xad000 + 4914440
	8   libsystem_pthread.dylib       	0x22a2f85a 0x22a2c000 + 14426
	9   libsystem_pthread.dylib       	0x22a2f7ce 0x22a2c000 + 14286
	10  libsystem_pthread.dylib       	0x22a2d724 0x22a2c000 + 5924
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 6 name:  AVAudioSession Notify Thread
	Thread 6:
	0   libsystem_kernel.dylib        	0x229728d0 0x22972000 + 2256
	1   libsystem_kernel.dylib        	0x229726d4 0x22972000 + 1748
	2   CoreFoundation                	0x22cbdac4 0x22c02000 + 768708
	3   CoreFoundation                	0x22cbbe4c 0x22c02000 + 761420
	4   CoreFoundation                	0x22c0b228 0x22c02000 + 37416
	5   CoreFoundation                	0x22c0b014 0x22c02000 + 36884
	6   libAVFAudio.dylib             	0x285ca224 0x28584000 + 287268
	7   libAVFAudio.dylib             	0x2859f176 0x28584000 + 110966
	8   libsystem_pthread.dylib       	0x22a2f85a 0x22a2c000 + 14426
	9   libsystem_pthread.dylib       	0x22a2f7ce 0x22a2c000 + 14286
	10  libsystem_pthread.dylib       	0x22a2d724 0x22a2c000 + 5924
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 7:
	0   libsystem_kernel.dylib        	0x229728d0 0x22972000 + 2256
	1   libsystem_kernel.dylib        	0x229726d4 0x22972000 + 1748
	2   Bunco Bonko                   	0x009a3d02 0xad000 + 9399554
	3   libsystem_pthread.dylib       	0x22a2f85a 0x22a2c000 + 14426
	4   libsystem_pthread.dylib       	0x22a2f7ce 0x22a2c000 + 14286
	5   libsystem_pthread.dylib       	0x22a2d724 0x22a2c000 + 5924
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 8 name:  com.apple.NSURLConnectionLoader
	Thread 8:
	0   libsystem_kernel.dylib        	0x229728d0 0x22972000 + 2256
	1   libsystem_kernel.dylib        	0x229726d4 0x22972000 + 1748
	2   CoreFoundation                	0x22cbdac4 0x22c02000 + 768708
	3   CoreFoundation                	0x22cbbe4c 0x22c02000 + 761420
	4   CoreFoundation                	0x22c0b228 0x22c02000 + 37416
	5   CoreFoundation                	0x22c0b014 0x22c02000 + 36884
	6   CFNetwork                     	0x2325c9f6 0x231d5000 + 555510
	7   Foundation                    	0x235254a4 0x23449000 + 902308
	8   libsystem_pthread.dylib       	0x22a2f85a 0x22a2c000 + 14426
	9   libsystem_pthread.dylib       	0x22a2f7ce 0x22a2c000 + 14286
	10  libsystem_pthread.dylib       	0x22a2d724 0x22a2c000 + 5924
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 9 name:  com.apple.CFSocket.private
	Thread 9:
	0   libsystem_kernel.dylib        	0x22986eec 0x22972000 + 85740
	1   CoreFoundation                	0x22cc2f12 0x22c02000 + 790290
	2   libsystem_pthread.dylib       	0x22a2f85a 0x22a2c000 + 14426
	3   libsystem_pthread.dylib       	0x22a2f7ce 0x22a2c000 + 14286
	4   libsystem_pthread.dylib       	0x22a2d724 0x22a2c000 + 5924
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 10:
	0   libsystem_kernel.dylib        	0x22987864 0x22972000 + 88164
	1   libsystem_pthread.dylib       	0x22a2db34 0x22a2c000 + 6964
	2   libsystem_pthread.dylib       	0x22a2d718 0x22a2c000 + 5912
Apr 18 22:55:39 Filippos-iPad ReportCrash[965] <Error>: Thread 0 crashed with ARM Thread State (32-bit):
	    r0: 0x1469d6c6    r1: 0x02519692      r2: 0x00000073      r3: 0x00000073
	    r4: 0x0000d7fb    r5: 0x0000ffff      r6: 0x000000bf      r7: 0x0251960c
	    r8: 0x00000001    r9: 0xffffffff     r10: 0x02519924     r11: 0x043c6000
	    ip: 0x1469c720    sp: 0x025195ec      lr: 0x0020705c      pc: 0x0057d1e8
	  cpsr: 0x20070010
```

**Comment by myflashlab** [comment]
> Same manifest file for iOS and Android

The new version has changed the manifest setup a bit as follow:  
When using ANELAB, this new changes should have been applied, please confirm that.
```
AIR Manifest changes:

    removed the tag <activity android:name="com.doitflash.facebook.access.MyLogin" ....
    removed the tag <activity android:name="com.doitflash.facebook.sharing.MyShare" ...
    removed the tag <activity android:name="com.doitflash.facebook.invite.MyInvite" ...
    Added <string>fb-messenger-share-api</string> and <string>fb-messenger</string> to LSApplicationQueriesSchemes.
```

**Comment by PippoApps** [comment]
Yep seems changes have been applied to manifest.
Here's the entire manifest:

```
<?xml version="1.0" encoding="utf-8" standalone="no" ?>
<application xmlns="http://ns.adobe.com/air/application/28.0">
  <id>com.yappingmoose.buncobonko</id>
  <versionNumber>0.9.7</versionNumber>
  <versionLabel/>
  <filename>Bunco Bonko</filename>
  <description/>
  <name>Bunco Bonko</name>
  <copyright/>
  <initialWindow>
    <content>BuncoBonko_Device.swf</content>
    <systemChrome>standard</systemChrome>
    <transparent>false</transparent>
    <visible>true</visible>
    <fullScreen>true</fullScreen>
    <aspectRatio>landscape</aspectRatio>
    <renderMode>gpu</renderMode>
    <autoOrients>true</autoOrients>
  </initialWindow>
  <icon>
    
    
    
    
    
    
  </icon>
  <customUpdateUI>false</customUpdateUI>
  <allowBrowserInvocation>false</allowBrowserInvocation>
  <iPhone>
    <requestedDisplayResolution>high</requestedDisplayResolution>
    <InfoAdditions><![CDATA[
  <key>CFBundleURLTypes</key>
  <array>
    <dict>
      <key>CFBundleTypeRole</key>
      <string>Editor</string>
      <key>CFBundleURLName</key>
      <string>google</string>
      <key>CFBundleURLSchemes</key>
      <array>
        <string>com.googleusercontent.apps.865990491833-nfb1bol4p8qmqqq8he4c96csgl5c1in2</string>
      </array>
    </dict>
    <dict>
      <key>CFBundleURLSchemes</key>
      <array>
        <string>fb970119236382852</string>
        <string>com.yappingmoose.buncobonko</string>
      </array>
    </dict>
  </array>
  <key>FacebookAppID</key>
  <string>970119236382852</string>
  <key>com.myflashlabs.firebase.deepLinkURLScheme</key>
  <string>com.yappingmoose.buncobonko</string>
  <key>UIDeviceFamily</key>
  <array>
    <string>1</string>
    <string>2</string>
  </array>
  <key>MinimumOSVersion</key>
  <string>9.0</string>
  <key>NSPhotoLibraryUsageDescription</key>
  <string>None</string>
  <key>LSApplicationQueriesSchemes</key>
  <array>
    <string>fbapi</string>
    <string>fb-messenger-api</string>
    <string>fb-messenger-share-api</string>
    <string>fb-messenger</string>
    <string>fbauth2</string>
    <string>fbshareextension</string>
  </array>
  <key>FacebookDisplayName</key>
  <string>Bunco Bonko</string>
  <key>com.myflashlabs.aneLab.id</key>
  <string>removed</string>
]]></InfoAdditions>
    <Entitlements><![CDATA[
  <key>application-identifier</key>
  <string>88VPSCX928.com.yappingmoose.buncobonko</string>
  <key>com.apple.developer.team-identifier</key>
  <string>88VPSCX928</string>
  <key>com.apple.developer.associated-domains</key>
  <array>
    <string>applinks:w322p.app.goo.gl</string>
  </array>
  <key>keychain-access-groups</key>
  <array>
    <string>88VPSCX928.*</string>
  </array>
  <key>aps-environment</key>
  <string>development</string>
  <key>get-task-allow</key>
  <true/>
]]></Entitlements>
  </iPhone>
  <android>
    <manifestAdditions><![CDATA[<manifest android:installLocation="auto" >
  <application>
    <activity>
      <intent-filter>
        <action android:name="android.intent.action.MAIN"/>
        <category android:name="android.intent.category.LAUNCHER"/>
      </intent-filter>
      <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <data android:scheme="http" android:host="www.playbuncoonline.com"/>
        <data android:scheme="https" android:host="www.playbuncoonline.com"/>
        <data android:scheme="com.yappingmoose.buncobonko"/>
      </intent-filter>
    </activity>
    <activity android:name="com.doitflash.inAppBilling.utils.MyPurchase" android:theme="@style/Theme.Transparent"/>
    <provider android:name="com.google.firebase.provider.FirebaseInitProvider" android:authorities="air.com.yappingmoose.buncobonko.firebaseinitprovider" android:exported="false" android:initOrder="100"/>
    <service android:name="com.myflashlab.firebase.core.MyFirebaseInstanceIdService" android:exported="true">
      <intent-filter>
        <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
      </intent-filter>
    </service>
    <meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version"/>
    <receiver android:name="com.google.firebase.iid.FirebaseInstanceIdReceiver" android:exported="true" android:permission="com.google.android.c2dm.permission.SEND">
      <intent-filter>
        <action android:name="com.google.android.c2dm.intent.RECEIVE"/>
        <action android:name="com.google.android.c2dm.intent.REGISTRATION"/>
        <category android:name="air.com.yappingmoose.buncobonko"/>
      </intent-filter>
    </receiver>
    <service android:name="com.google.firebase.iid.FirebaseInstanceIdService" android:exported="true">
      <intent-filter android:priority="-500">
        <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
      </intent-filter>
    </service>
    <activity android:name="com.google.android.gms.common.api.GoogleApiActivity" android:theme="@android:style/Theme.Translucent.NoTitleBar" android:exported="false"/>
    <meta-data android:name="com.facebook.sdk.ApplicationId" android:value="\ 970119236382852"/>
    <activity android:name="com.facebook.FacebookActivity" android:configChanges="keyboard|keyboardHidden|screenLayout|screenSize|orientation" android:theme="@android:style/Theme.Translucent.NoTitleBar" android:label="Bunco Bonko"/>
    <activity android:name="com.facebook.CustomTabMainActivity"/>
    <activity android:name="com.facebook.CustomTabActivity" android:exported="true">
      <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <data android:scheme="fb970119236382852"/>
      </intent-filter>
    </activity>
    <provider android:authorities="com.facebook.app.FacebookContentProvider970119236382852" android:name="com.facebook.FacebookContentProvider" android:exported="true"/>
    <meta-data android:name="com.myflashlabs.aneLab.id" android:value="removed"/>
  </application>
  <uses-sdk android:minSdkVersion="19" android:targetSdkVersion="23"/>
  <uses-permission android:name="com.android.vending.BILLING"/>
  <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
  <uses-permission android:name="com.google.android.c2dm.permission.RECEIVE"/>
  <uses-permission android:name="air.com.yappingmoose.buncobonko.permission.C2D_MESSAGE"/>
  <uses-permission android:name="android.permission.WAKE_LOCK"/>
  <uses-permission android:name="android.permission.INTERNET"/>
</manifest>]]></manifestAdditions>
  </android>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <extensions>
    <extensionID>com.myflashlab.air.extensions.billing</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.androidSupport</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.analytics.impl</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.common</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.crash</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.dynamicLinks</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.iid</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.appinvite</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.basement</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.base</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.tasks</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.overrideAir</extensionID>
    <extensionID>com.myflashlab.air.extensions.facebook</extensionID>
    <extensionID>com.myflashlab.air.extensions.firebase.core</extensionID>
    <extensionID>com.myflashlab.air.extensions.firebase.crash</extensionID>
    <extensionID>com.myflashlab.air.extensions.firebase.dynamicLinks</extensionID>
    <extensionID>com.myflashlab.air.extensions.firebase.invites</extensionID>
  </extensions>
</application>
```

**Comment by PippoApps** [comment]
Upgraded to AIR 29, rerun all setups in AneLab, same behavior.

**Comment by PippoApps** [comment]
Same exact behavior on iOS 10.3.3
```
428 [ANEFacebook]		Initialized facebook app 970119236382852
FB_ANE> checkOverride_openURLOptions
FB_ANE> com.yappingmoose.buncobonko://google/link/?dismiss=1&is_weak_match=1
```

**Comment by myflashlab** [comment]
This seems like a conflict between Firebase dynamiclinks and facebook. We are now actively testing this. we'll have more news on this soon.

**Comment by myflashlab** [comment]
Ok, after testing and testing this, we cannot duplicate the problem no matter what we do. it's just working fine! can you somehow provide more information?

I also tried to google "dismiss=1&is_weak_match=1" check this out: https://www.google.com/search?q=%22dismiss%3D1%26is_weak_match%3D1%22&ie=utf-8&oe=utf-8&client=firefox-b-ab

can you relate anything?

**Comment by PippoApps** [comment]
Thank yuo, it seems that's a default link when there is no dynamic link. So that's not related to facebook. Still, facebook does not trigger any init. I will try without initializing firebase.

**Comment by PippoApps** [comment]
Updated to latest AneLab and all ANEs.
Still, initializing Facebook does not open facebook login window and does not trigger any callback :(

**Comment by PippoApps** [comment]
Not sure if this can help, traces shortly after initializing facebook:
Apr 25 01:12:47 Filippos-iPad com.apple.xpc.launchd[1] (com.apple.xpc.launchd.domain.pid.Bunco Bonko.1173) <Error>: Caller not allowed to perform action: Bunco Bonko.1173, action = kill service, code = 154: Requestor is not a platform binary, uid = 501, euid = 501, gid = 501, egid = 501, asid = 0
Apr 25 01:12:47 Filippos-iPad Bunco Bonko[1173] <Error>: Could not signal service com.apple.WebKit.WebContent: 154: Requestor is not a platform binary
Apr 25 01:12:47 Filippos-iPad Bunco Bonko[1173] <Error>: Could not signal service com.apple.WebKit.Networking: 154: Requestor is not a platform binary

**Comment by PippoApps** [comment]
Same behavior on iOS 10.3.3

**Comment by PippoApps** [comment]
Tried on 2 different devices (iPad 3 with 9.3.5 and iPhone 5 with 10.3.3), before and after factory reset, with or without facebook app installed. Behaviour is exactly the same. Nothing happens when Facebook is initialized. Same code works on Android.
Can you point me in some direction? Might something be missing in AIR framework?

**Comment by PippoApps** [comment]
Even avoiding Firebase initialization still nothing happens.

**Comment by PippoApps** [comment]
Tried manually reinstalling frameworks downloaded from here. To no avail.
Still same behavior. Absolutely nothing happens when calling Facebook.init("...");
Can you point me in some direction?

**Comment by PippoApps** [comment]
Same reverting to AIR 28. I am positive it's the ANE at this point...

**Comment by myflashlab** [comment]
Your setup seems all fine :thinking: One thing to check is that when are you calling the ```Facebook.init(```? it should be in the documentClass and very soon when your app starts. can you confirm that?

Besides, does it work in a simple empty project having the facebook ANE on it only? The first thing I would do to figure out the problem is to create a new project with the same package name which have the facebook app already setup in facebook developer account and see if I can make that work. Have you tried that? I'm pretty sure there's something wrong in your project setup because Facebook ANE is one of our most used ANEs and others don't seem to have a similar problem and we are not able to duplicate the problem neither. However, we are still doing more tests on this. While doing that, please check the above suggestions and tell us what you see.

**Comment by myflashlab** [comment]
Ok, we got it! The reason must be because you are initializing the Facebook ANE too late in your project. Here's what we just did: in a test project we initialized firebase with DynamicLinks first and then tried to initialize the Facebook after a few seconds. This works on Android but not on iOS. As mentioned in wiki, you must init Facebook ANE as soon as possible because facebook is listening to app launch event by iOS and if you initialize it too late in your app, the launching event will be fired by iOS native and the ANE is not available to catch it!

FYI: by default AdobeAir apps are not able to catch iOS app launching events at all but in OverrideAir ANE, we have done some tricks to be able to catch it anyway.

**Comment by PippoApps** [comment]
Thank you very much. Nope it does not initialize in the document constructor but only after having retrieved some initialization data. It used to work with previous ANE that's why I didn't think of it :) I'll check that and come back to you.

**Comment by myflashlab** [comment]
You must have been using the VERY old version before OR you may have not been using the DynamicLink ANE before.

The logic about how we are catching the iOS app-launching event which AdobeAIR can't catch by default without an ANE:
1. Override ANE will be initialized automatically even before the AdobeAIR app starts and listens to the iOS app-launching events and some more.
2. Override ANE will be hit with those events but AIR is not yet up, so it keeps the events in a Dictionary.
3. When AIR starts and ANEs are initialized, OverrideAIR ANE will pass the events to all the ANEs that are expecting it.

If there's only one ANE expecting the event, or if there are 10 ANEs expecting it, they will all receive it. But if an ANE (facebook in your case) is initialize too late, then overrideAIR has already delivered the event and the late ANE will never be able to catch that event.

Yeah, iOS is super hard to be dealt with :)

**Comment by PippoApps** [comment]
Thank for the insight! Very useful!

**Comment by PippoApps** [comment]
I am rebuilding the entire initialization flow since it relied on a lot of configuration data. One question: how about Billing ANE? I can't initialize with product IDs until I received the list of active items from server, will I be able to perform initialization afterwards?

**Comment by PippoApps** [comment]
Yeah :) facebook login is now triggered.
An old weird bug came back though, I'll report in another issue.
Thank you very much.

**Comment by myflashlab** [comment]
> One question: how about Billing ANE? I can't initialize with product IDs until I received the list of active items from server, will I be able to perform initialization afterwards?

I think yes, unlike FB and DynamicLinks, the IAP ANE is not expecting those app-launching events and it should be fine.
